src/store/d-apps/d-apps-epics.ts

```diff
@@ -37,18 +37,33 @@ const removePermissionEpic = (action$: Observable<Action>) =>
   action$.pipe(
     ofType(removePermissionAction),
     toPayload(),
-    switchMap(accountIdentifier =>
-      from(BeaconHandler.removePermission(accountIdentifier)).pipe(
-        map(() => {
-          showSuccessToast('Permission successfully removed!');
-
-          return loadPermissionsActions.submit();
-        }),
-        catchError(err => {
-          showErrorToast(err.message);
-
-          return EMPTY;
-        })
+    switchMap(({ accountIdentifier, senderId }) =>
+      from(BeaconHandler.getPeers()).pipe(
+        switchMap(peers =>
+          forkJoin(
+            peers.map(async peer => {
+              const peerSenderId = await getSenderId(peer.publicKey);
+              if (senderId === peerSenderId) {
+                return BeaconHandler.removePeer({ ...peer, senderId: peerSenderId });
+              }
+            })
+          ).pipe(
+            switchMap(() =>
+              from(BeaconHandler.removePermission(accountIdentifier)).pipe(
+                map(() => {
+                  showSuccessToast('Permission successfully removed!');
+
+                  return loadPermissionsActions.submit();
+                }),
+                catchError(err => {
+                  showErrorToast(err.message);
+
+                  return EMPTY;
+                })
+              )
+            )
+          )
+        )
```

Comment:
First of all, try not to mix `observalbes`/`rxjs` with `promises`/`async & await` approaches.
```
forkJoin(
            peers.map(async peer => {
              const peerSenderId = await getSenderId(peer.publicKey);
              if (senderId === peerSenderId) {
                return BeaconHandler.removePeer({ ...peer, senderId: peerSenderId });
              }
            })
          )
```
Change to this: 
```
          forkJoin(
            peers.map(peer =>
              from(getSenderId(peer.publicKey)).pipe(
                switchMap(peerSenderId =>
                  senderId === peerSenderId ? BeaconHandler.removePeer({ ...peer, senderId }) : EMPTY
                )
              )
            )
          )
```
Secondly, try to avoid redundant `.pipe()`s.   
You should get something like this: 
```
const removePermissionEpic = (action$: Observable<Action>) =>
  action$.pipe(
    ofType(removePermissionAction),
    toPayload(),
    switchMap(({ accountIdentifier, senderId }) =>
      from(BeaconHandler.getPeers()).pipe(
        switchMap(peers =>
          forkJoin(
            peers.map(peer =>
              from(getSenderId(peer.publicKey)).pipe(
                switchMap(peerSenderId =>
                  senderId === peerSenderId ? BeaconHandler.removePeer({ ...peer, senderId }) : EMPTY
                )
              )
            )
          )
        ),
        switchMap(() => BeaconHandler.removePermission(accountIdentifier)),
        map(() => {
          showSuccessToast('Permission successfully removed!');

          return loadPermissionsActions.submit();
        }),
        catchError(err => {
          showErrorToast(err.message);

          return EMPTY;
        })
      )
    )
  );
```